### PR TITLE
feat: add _getTextboxValueFromOption method

### PR DESCRIPTION
**Pull request by MatthieuLebigre**
## What I did

This PR aims to fix this discussion: https://github.com/ing-bank/lion/discussions/1247

### CODE REVIEW

File: packages/combobox/src/LionCombobox.js [hunk 1]
```diff
@@ -850,7 +870,11 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
    */
   _syncToTextboxMultiple(modelValue, oldModelValue = []) {
     const diff = modelValue.filter(x => !oldModelValue.includes(x)).toString();
-    this._setTextboxValue(diff); // or last selected value?
+    const newValue = this.formElements
+      .filter(option => diff.includes(option.choiceValue))
+      .map(option => this._getTextboxValueFromOption(option))
+      .join(' ');
```
Comment (jorenbroekema) [inline packages/combobox/src/LionCombobox.js:876]: So multiple choices are concatenated with a space? Can we add a test for that?

File: packages/combobox/src/LionCombobox.js [hunk 2]
```diff
@@ -450,6 +460,17 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
...
   }
 
+  /**
+   * Return the value to be used for the input value
+   * @overridable
+   * @param {LionOption} option
+   * @returns {string}
+   */
+  // eslint-disable-next-line class-methods-use-this
+  _getTextboxValueFromOption(option) {
```
Comment (jorenbroekema) [inline packages/combobox/src/LionCombobox.js:470]: Can we add some tests for this method and the expected behaviors when you override this method?

File: packages/combobox/src/LionCombobox.js [hunk 3]
```diff
@@ -335,8 +340,12 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
    */
   __unsyncCheckedIndexOnInputChange() {
     const autoselect = this._autoSelectCondition();
-    if (!this.multipleChoice && !autoselect && !this._inputNode.value.startsWith(this.modelValue)) {
-      this.checkedIndex = -1;
+    const checkedElement = this.formElements[/** @type {number} */ (this.checkedIndex)];
+    if (!this.multipleChoice && !autoselect && checkedElement) {
```
Comment (jorenbroekema) [inline packages/combobox/src/LionCombobox.js:344]: If checkedElement does not exist, should the checkedIndex be set to -1 as well? That edge case doesn't seem covered by the logic here, but I could be wrong as no tests are failing.

**Review by jorenbroekema — DISMISSED** [review]
Mostly just adding tests and running `npm run format` so prettier doesn't complain :). Thanks for creating the PR!

File: packages/combobox/src/LionCombobox.js (hunk 3 above)
Comment (MatthieuLebigre) [reply packages/combobox/src/LionCombobox.js:344]: I believe if checkedElement doesn't exist, checkedIndex should already be set to -1. I don't think that's the place where it should set ?

File: packages/combobox/src/LionCombobox.js (hunk 2 above)
Comment (MatthieuLebigre) [reply packages/combobox/src/LionCombobox.js:470]: done

File: packages/combobox/src/LionCombobox.js (hunk 1 above)
Comment (MatthieuLebigre) [reply packages/combobox/src/LionCombobox.js:877]: I've added some tests. But I'm not so sure to understand what is to be done here. I've copy/paste the code from the discussion https://github.com/ing-bank/lion/discussions/1247

**Comment by MatthieuLebigre** [comment]
> Mostly just adding tests and running `npm run format` so prettier doesn't complain :). Thanks for creating the PR!

For prettier, when I run `npm run format` it modifies a bunch of files, but not the one I've changed :D Have you already experienced that?

**Comment by tlouisse** [comment]
> > Mostly just adding tests and running `npm run format` so prettier doesn't complain :). Thanks for creating the PR!
> 
> For prettier, when I run `npm run format` it modifies a bunch of files, but not the one I've changed :D Have you already experienced that?

It appears that different configs of the Prettier vscode plugin can explain those differences (even though they take .eslintrc, prettier config in package.json into consideration.) 
Not sure what 'wins' and how we can best mitigate this problem

**Comment by MatthieuLebigre** [comment]
> > > Mostly just adding tests and running `npm run format` so prettier doesn't complain :). Thanks for creating the PR!
> > 
> > 
> > For prettier, when I run `npm run format` it modifies a bunch of files, but not the one I've changed :D Have you already experienced that?
> 
> It appears that different configs of the Prettier vscode plugin can explain those differences (even though they take .eslintrc, prettier config in package.json into consideration.)
> Not sure yet what 'wins' and when and how we can best mitigate this problem

I tried a lot of different options/configuration but it doesn't work. I tried turning off the VSCode plugin, changing the config to a `.prettierrc` file, but it changes nothing :(

File: packages/combobox/src/LionCombobox.js (hunk 1 above)
Comment (tlouisse) [reply packages/combobox/src/LionCombobox.js:877]: Before, I think setting an array to the textbox implicitly had the same behavior (concatting with spaces). This makes or  more specific and type compatible. But an extra test never hurts ;)
Maybe you can disable Prettier plugin (on autosave) and add your new tests again?

File: packages/combobox/src/LionCombobox.js (hunk 3 above)
Comment (tlouisse) [reply packages/combobox/src/LionCombobox.js:344]: Should indeed be the case that checkedEl will only not be found if index is already -1

**Review by tlouisse — CHANGES_REQUESTED** [review]
Maybe you can reapply the tests with prettier disabled? Then we can review the tests better.
The formatting inconsistencies could be tackled in a separate issue.

File: packages/combobox/src/LionCombobox.js (hunk 1 above)
Comment (MatthieuLebigre) [reply packages/combobox/src/LionCombobox.js:876]: Should be good now for the format. I did some investigation and will open an issue to discuss about it :)

File: packages/combobox/test/lion-combobox.test.js [hunk 4]
```diff
@@ -1575,6 +1575,211 @@ describe('lion-combobox', () => {
       });
     });
 
+    describe('Subclassers', () => {
```
Comment (tlouisse) [inline packages/combobox/test/lion-combobox.test.js:1578]: Nice :+1: 
Thanks a lot.

Could you maybe rename this describe to `'Override _getTextboxValueFromOption'`  and put it inside the Subclassers describe above?

If @jorenbroekema also agrees, it only needs:
- a reformat of this file, so that it passes the ci
- a changeset 
- a squash of the commits

Comment (MatthieuLebigre) [reply packages/combobox/test/lion-combobox.test.js]: - `describe` is moved (be aware there are already several `describe('Subclassers'` in this file)
- changeset added
- squash done
- For the format, unfortunately I can't do it, I'm not able to find the proper way to make prettier work (see https://github.com/ing-bank/lion/issues/1388)

Comment (tlouisse) [reply packages/combobox/test/lion-combobox.test.js]: Good catch about the multiple 'Subclassers' describes :) We will wrap up this mr and fix that later.

Prettier seems to work now (after a rebase with latest master, which pobably uses a later Prettier version).

**Comment by tlouisse** [comment]
Everything is solved 👍 Thanks for this contribution

**Review by tlouisse — APPROVED** [review]
